### PR TITLE
[Doc] Document undefined symbol error with torch version < 1.13

### PR DESCRIPTION
**Pull request by nickspell**
## Description
This PR:
- Documents an issue with PyTorch versioning
- Proposes some workarounds to make the library work with older PyTorch versions installed.
- Links such documentation in the troubleshooting section of the README

## Motivation and Context

When using torch at lower version than 1.13 and installing the package stable version via pip, an undefined symbol error appears when trying to import the module. 

Related to https://github.com/pytorch/rl/issues/689.

## Types of changes
- [x] Documentation (update in the documentation)

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: knowledge_base/VERSIONING_ISSUES.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+### How to reproduce
+1. Create an Colab Notebook (at 24/11/2022 Colab enviroment has Python 3.7 and Pytorch 1.12 installed by default).
+2. ``` !pip install torchrl ```
+3. ``` import torchrl ```
+
+### Workarounds
+1. Install torch 1.13 or above.
+2. Install latest version of functorch. This version depends on torch 1.13 and will install it for you.
```
Comment (vmoens) [inline knowledge_base/VERSIONING_ISSUES.md:18]: Here I would be a bit more precise:
If you install torch >= 1.13, functorch will be part of the package
Otherwise (for torch <=1.12.1) you should install functorch via `pip install functorch==0.2.0`

If users choose to select (2) they will need to install from source anyway (as specified in (3))

So i think we should say:

> There are two workarounds to this issue:
> 1. Install/upgrade to the latest pytorch
> 2. Install functorch and install the library from source

File: knowledge_base/VERSIONING_ISSUES.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Versioning Issues
+
+## Pytorch version
+This issue is related to https://github.com/pytorch/rl/issues/689. Using PyTorch versions lower < 1.13 and installing stable package leads to the following error:
```
Comment (vmoens) [inline knowledge_base/VERSIONING_ISSUES.md:4]: I guess that in the future we could have other undefined symbol, maybe we want to smooth things a bit (e.g.
> ```
> undefined symbol _ZN8pybind116detail11type_casterIN2at6TensorEvE4loadENS_6handleEb
> ```
> or another symbol

)

File: README.md [hunk 3]
```diff
@@ -506,6 +506,10 @@ and not
 OS: macOS **** (x86_64)
 ```
 
+If you successfully install and you have a ```ImportError: /usr/local/lib/python3.7/dist-packages/torchrl/_torchrl.so: undefined symbol: _ZN8pybind116detail11type_casterIN2at6TensorEvE4loadENS_6handleEb``` when you try to import the module, most probably this is caused by an incompatibility with your torch package.
```
Comment (vmoens) [inline README.md:509]: Maybe rephrase such that we can add things in the versioning issues doc:

> Versioning issues can cause error message of the type `undefined symbol` and such. For these, refer to the [versioning issues dedicated document](knowledge_base/VERSIONING_ISSUES.md) for a complete explanation.

**Review by vmoens — COMMENTED** [review]
We could maybe make this a bit more generic? See my suggestions

File: knowledge_base/VERSIONING_ISSUES.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+This is probably due to some incompatibility (tensor casting?) of torch C++ bindings between version 1.12 and 1.13. 
+
+
+### How to reproduce
+1. Create an Colab Notebook (at 24/11/2022 Colab enviroment has Python 3.7 and Pytorch 1.12 installed by default).
+2. ``` !pip install torchrl ```
+3. ``` import torchrl ```
+
+### Workarounds
+1. Install torch 1.13 or above.
```
[the remark this answers is not in the document]
Comment (nickspell) [reply knowledge_base/VERSIONING_ISSUES.md]: Added line at the end of how to reproduce section to address quick fix for the Colab example

File: README.md (hunk 3 above)
Comment (nickspell) [reply README.md]: rephrased as suggested

File: knowledge_base/VERSIONING_ISSUES.md (hunk 2 above)
Comment (nickspell) [reply knowledge_base/VERSIONING_ISSUES.md]: Rephrased to make it clear that such specific error is just an example

**Review by vmoens — APPROVED** [review]
LGTM!
